treeherder/webapp/api/jobs.py

```diff
@@ -37,50 +114,142 @@ class JobsViewSet(viewsets.ViewSet):
+        ('build_os', 'build_platform__os_name', None),
+        ('build_platform', 'build_platform__platform', None),
+        ('build_platform_id', 'build_platform_id', None),
+        ('build_system_type', 'signature__build_system_type', None),
+        ('end_timestamp', 'end_time', to_timestamp),
+        ('failure_classification_id', 'failure_classification_id', None),
+        ('id', 'project_specific_id', None),
+        ('job_coalesced_to_guid', 'coalesced_to_guid', None),
+        ('job_group_description', 'job_type__job_group__description', None),
+        ('job_group_id', 'job_type__job_group_id', None),
+        ('job_group_name', 'job_type__job_group__name', None),
+        ('job_group_symbol', 'job_type__job_group__symbol', None),
+        ('job_guid', 'guid', None),
+        ('job_type_description', 'job_type__description', None),
+        ('job_type_id', 'job_type_id', None),
+        ('job_type_name', 'job_type__name', None),
+        ('job_type_symbol', 'job_type__symbol', None),
+        ('last_modified', 'last_modified', None),
+        ('machine_name', 'machine__name', None),
+        ('machine_platform_architecture', 'machine_platform__architecture', None),
+        ('machine_platform_os', 'machine_platform__os_name', None),
+        ('option_collection_hash', 'option_collection_hash', None),
+        ('platform', 'machine_platform__platform', None),
+        ('push_id', 'push_id', None),
+        ('reason', 'reason', None),
+        ('ref_data_name', 'signature__name', None),
+        ('result', 'result', None),
+        ('result_set_id', 'push_id', None),
+        ('running_eta', 'running_eta', None),
+        ('signature', 'signature__signature', None),
+        ('start_timestamp', 'start_time', to_timestamp),
+        ('state', 'state', None),
+        ('submit_timestamp', 'submit_time', to_timestamp),
+        ('tier', 'tier', None),
+        ('who', 'who', None),
+    ]
+
+    _option_collection_hash_idx = [pq[0] for pq in _property_query_mapping].index(
+        'option_collection_hash')
+
+    def _get_option_collection_map(self):
         option_collection_map = {}
         for (hash, option_name) in OptionCollection.objects.values_list(
                 'option_collection_hash', 'option__name'):
             if not option_collection_map.get(hash):
                 option_collection_map[hash] = option_name
             else:
                 option_collection_map[hash] += (' ' + option_name)
-
         return option_collection_map
 
-    @with_jobs
-    def retrieve(self, request, project, jm, pk=None):
+    def _get_job_list_response(self, job_qs, offset, count, return_type):
+        '''
+        custom method to serialize + format jobs information
+
+        It's worth doing this big ugly thing (as opposed to using
+        the django rest framework serializer or whatever) as
+        this function is often in the critical path
+        '''
+        option_collection_map = self._get_option_collection_map()
```

Comment:
I think just because it's less trouble. we only have 4 or so values for option collection at the moment.